dependency_graph/generate.sh

```diff
@@ -0,0 +1 @@
+python ./generate.py | tred | dot -Tpng > dependency_graph.png
```

Comment:
Perhaps switch to `-Tsvg` instead? I think that an svg is much more readable and navigateable when you take into account the size of the generated graph.